### ISSUE TITLE
Light sources

### ISSUE THREAD

**Issue by Dany0** [iamgreaser/iceball#165]
The engine should support light sources.

**Comment by iamgreaser** [comment]
It should support shaders first.

**Comment by iamgreaser** [comment]
Yes, that's right, this is all OpenGL 1.1 + GL_ARB_vertex_buffer_object. Thus, we have no shaders.

I'll be fine with adding GL 2.1 shaders, but will need to work out the best approach for it, and will also have to stab people who make shaders that only works on their GPU.
